app/controllers/internal/events_controller.rb

```diff
@@ -0,0 +1,140 @@
+module Internal
+  class EventsController < ::InternalController
+    before_action :load_buildings, only: %i[new edit create update]
+    layout "internal"
+
+    def index
+      @no_results = load_pending_events.blank?
+    end
+
+    def show
+      @event = GetIntoTeachingApiClient::TeachingEventsApi.new.get_teaching_event(params[:id])
+      @page_title = @event.name
+      if @event.status_id != GetIntoTeachingApiClient::Constants::EVENT_STATUS["Pending"]
+        raise ActionController::RoutingError, "Not Found"
+      end
+    end
+
+    def new
+      @event = Event.new
+      @event.building = EventBuilding.new
+      @event.building.fieldset = "existing"
+    end
+
+    def final_submit
+      event = GetIntoTeachingApiClient::TeachingEventsApi.new.get_teaching_event(params[:format])
+      @event = transform_event(event)
```

Comment:
I would be inclined to move the mapping to/from an API event to your local event type into the `Internal::Event` so you have something like:

```
@event = Internal::Event.new(api_event)
...
@api_event = @internal_event.to_api_event
```

You could then unit test it